documentation/src/main/asciidoc/topics/ref_performance_tuning.adoc

```diff
@@ -79,6 +79,78 @@ The recommended values are:
 * send window size - 640 KB
 * receive window size - 25 MB
 
+== Thread pool settings
+
+A thread pool manages threads by assigning them concurrent tasks.
+Additionally, a thread pool can improve system performance by using any idle thread instead of creating threads.
+
+Thread pools limit the creation and deletion of threads to active worker threads, so that the amount of threads does not typically exceed the amount of tasks.
+This helps reduce the memory footprint of your {brandname} server.
+
+Your {brandname} configuration applies default settings to thread pools, but you can change these settings to meet your needs.
+
+The following table outlines common thread pool properties and their default settings:
+
+[cols="1,1"]
+|===
+| Property | Default setting | Description
+
+|`ASYNC_NOTIFICATION_EXECUTOR` |`1`| Used by asynchronous listeners that your {brandname} server registers in embedded mode. 
+The default setting ensures that Infinispan invokes asynchronous listeners according to importance of their messages. 
+Configuring this property is discouraged.
+Increasing the property's value from `1` might cause your {brandname} server to randomly create mixed types of listener events on different threads.
+|`BLOCKING_EXECUTOR` | `150` |  Used by your {brandname} server to handle blocking operations.
+An example of a blocking operation includes a cache store that calls external sources, transactions, or time-consuming non-blocking entities.
+Thread pool reuses blocking threads based on the value specified for this property.
+For workloads with large amounts of blocking threads, you set a higher value for the `BLOCKING_EXECUTOR` property.
+|`EXPIRATION_SCHEDULED_EXECUTOR`|`1`| Based on the property's value, a reaper schedules checks on expired cache entries that exist in memory.
+If your workload includes a large amount of expired cache entries that span multiple caches, consider setting a higher value for this property. 
+Be mindful that setting a higher value does not increase the rate at which threads process expired cache entries.
+
+|`NON_BLOCKING_EXECUTOR` | `ProcessorInfo.availableProcessors() * 2`| Property defines how a thread pool handles server operations and embedded cache operations. For example, 
+a thread pool sends a blocking operation to a non-blocking thread.
+Be mindful that a thread pool does not use this property when processing client threads.
+Configuring this property is discouraged, because the default setting scales thread workload according to the size of your server installation, which ensures lower latency for thread operations.
+If you have additional CPU cores in an {k8s} environment, consider increasing the core CPU value on the basis that you need to reduce non-blocking threads. 
```

Comment:
This sounds a bit confusing to me

Reply:
Hi @wfink . Would you be OK to elaborate further? The complete table entry is confusing or just a sentence?

The use case you mentioned was based around Kubernetes? CPU cores refer to an amount of just an amalgamation of one resource?

Reply:
I moved an updated note to the `non-blocking-bounded-queue-thread-pool` table entry. WDYT?

> Note: If you have a large amount of running CPU cores in your environment, your JVM detects these cores and creates a high amount of threads. {brandname} creates more non-blocking thread pools to manage the threads. These activities consume large amounts of caching resources.